tools/walletextension/storage/sqlite.go

```diff
@@ -0,0 +1,113 @@
+package storage
+
+import (
+	"database/sql"
+	"encoding/hex"
+	"fmt"
+
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/ethereum/go-ethereum/crypto"
+	"github.com/ethereum/go-ethereum/crypto/ecies"
+	_ "github.com/mattn/go-sqlite3" // sqlite driver for sql.Open()
+	"github.com/obscuronet/go-obscuro/go/rpc"
+)
+
+type SqliteDatabase struct {
+	db       *sql.DB
+	hostAddr string
+}
+
+func NewSqliteDatabase(dbName string, hostAddr string) *SqliteDatabase {
+	db, err := sql.Open("sqlite3", dbName)
+	if err != nil {
+		fmt.Println("Error opening datbaase: ", err)
+		return nil
+	}
+
+	_, err = db.Exec(`CREATE TABLE IF NOT EXISTS viewingkeys (
+		user_id TEXT,
+		host_address TEXT,
```

Comment:
not sure I follow here.
What is the purpose of the host_address?

Reply:
host_address is used in persistence, I removed it from my first implementation, but now tried if it would help with failing test (it doesn't). Removed it now.